### PR TITLE
fix!: issues related to inconsistent parsing and generation of `CLUSTER BY`

### PR DESCRIPTION
**Pull request by JoshuaM176**
Cluster By was parsing differently for creates and alters which caused the output to be incorrect.
Here the result from the alter does not include the Paren expression, but the create does.
sqlglot.parse_one("alter table test cluster by (col1)")
sqlglot.parse_one("create table test cluster by (col1)")

Using the Ordered expression was also causing problems since transpiling between dialects with different defaults for nulls_first would result in the null ordering showing up in the output, but cluster by does not support ordering.

The main changes I made include parsing the cluster by as wrapped whenever it has a ( following it and using a wrapped property to track if the cluster should have parens, as well as using _parse_column rather than _parse_ordered.

### CODE REVIEW

**Comment by JoshuaM176** [comment]
Made some further improvements, split cluster into two different expressions, ClusterProperty for creates/alters and Cluster for queries. And covered big query where parentheses are not used in the create statement Cluster By so transpiling to/from bigquery will now correctly add/exclude parentheses

File: sqlglot/expressions/query.py [hunk 1]
```diff
@@ -859,8 +859,8 @@ class SkipJSONColumn(Expression):
     arg_types = {"regexp": False, "expression": True}
 
 
-class Cluster(Order):
-    pass
+class Cluster(Expression):
+    arg_types = {"expressions": True}
```
Comment (georgesittas) [inline sqlglot/expressions/query.py:863]: @geooo109 just a heads up that I checked out the docs and did some quick testing. This seems to be legit, I think the previous parsing behavior was based on erroneous assumptions. FYI, it was added ~3 years ago.

So, changes like the example in the docstring below are ok.

File: sqlglot/parser.py [hunk 2]
```diff
@@ -3064,15 +3064,17 @@ def _parse_checksum(self) -> exp.ChecksumProperty:
...
-                    self._parse_wrapped_csv(self._parse_ordered)
-                    if wrapped
-                    else self._parse_csv(self._parse_ordered)
-                )
-            )
+    def _parse_cluster(self) -> exp.Cluster:
+        self._match(TokenType.CLUSTER_BY)
+        return exp.Cluster(
+            expressions=self._parse_csv(self._parse_column),
+        )
```
Comment (geooo109) [inline sqlglot/parser.py:3073]:
```suggestion
def _parse_cluster(self) -> exp.Cluster:
    self._match(TokenType.CLUSTER_BY)
    return self.expression(
        exp.Cluster,
        expressions=self._parse_csv(self._parse_column),
    )
```

File: sqlglot/parser.py [hunk 3]
```diff
@@ -3064,15 +3064,17 @@ def _parse_checksum(self) -> exp.ChecksumProperty:
...
+
+    def _parse_cluster_property(self) -> exp.ClusterProperty:
+        if self._match_texts(("AUTO", "NONE"), advance=False):
+            return exp.ClusterProperty(this=self._parse_var())
+        return exp.ClusterProperty(
+            expressions=self._parse_wrapped_csv(self._parse_column),
         )
```
Comment (geooo109) [inline sqlglot/parser.py:3080]:
```suggestion
  def _parse_cluster_property(self) -> exp.ClusterProperty:
      if self._match_texts(("AUTO", "NONE")):
          return self.expression(exp.ClusterProperty, this=exp.var(self._prev.text.upper()))
     
      return self.expression(
          exp.ClusterProperty, expressions=self._parse_wrapped_csv(self._parse_column)
      )

```

File: sqlglot/parsers/bigquery.py [hunk 4]
```diff
@@ -509,6 +509,11 @@ def _parse_column(self) -> exp.Expr | None:
 
         return column
 
+    def _parse_cluster_property(self) -> exp.ClusterProperty:
+        return exp.ClusterProperty(
+            expressions=self._parse_csv(self._parse_column),
+        )
```
Comment (geooo109) [inline sqlglot/parsers/bigquery.py:517]:
```suggestion
def _parse_cluster_property(self) -> exp.ClusterProperty:
    return self.expression(
        exp.ClusterProperty,
        expressions=self._parse_csv(self._parse_column),
    )
```

File: sqlglot/generator.py [hunk 5]
```diff
@@ -3042,6 +3043,11 @@ def withfill_sql(self, expression: exp.WithFill) -> str:
     def cluster_sql(self, expression: exp.Cluster) -> str:
         return self.op_expressions("CLUSTER BY", expression)
 
+    def clusterproperty_sql(self, expression: exp.ClusterProperty) -> str:
+        if expression.args.get("this"):
+            return f"CLUSTER BY {self.sql(expression, 'this')}"
+        return f"CLUSTER BY ({self.expressions(expression, flat=True)})"
```
Comment (geooo109) [inline sqlglot/generator.py]:
```suggestion
def clusterproperty_sql(self, expression: exp.ClusterProperty) -> str:
    this = self.sql(expression, "this") or f"({self.expressions(expression, flat=True)})"
    return f"CLUSTER BY {this}"
```

File: sqlglot/generators/bigquery.py [hunk 6]
```diff
@@ -688,3 +688,8 @@ def cast_sql(self, expression: exp.Cast, safe_prefix: str | None = None) -> str:
                 return f"{self.sql(expression, 'to')}{self.sql(this)}"
 
         return super().cast_sql(expression, safe_prefix=safe_prefix)
+
+    def clusterproperty_sql(self, expression: exp.ClusterProperty) -> str:
+        if expression.args.get("this"):
+            return ""
+        return self.op_expressions("CLUSTER BY", expression)
```
Comment (geooo109) [inline sqlglot/generators/bigquery.py:696]:
```
def clusterproperty_sql(self, expression: exp.ClusterProperty) -> str:
    if expression.args.get("this"):
        self.unsupported(f"Unsupported CLUSTER BY {self.sql(expression, 'this')}")
        return ""
    return self.op_expressions("CLUSTER BY", expression)
```

**Review by geooo109 — COMMENTED** [review]
Nice work! @JoshuaM176 

Let's also add tests for `CLUSTER BY AUTO/NONE` and also for `CREATE/ALTER` with multi columns.

**Comment by JoshuaM176** [comment]
Pushed changes adding the self.expression(...) and other style suggestion. I also removed AUTO/NONE from the generic parser and moved it into databricks since databricks is the only dialect that supports that. And made it unsupported for the generic generator so databricks can specifically override that. Still working on adding tests

**Comment by JoshuaM176** [comment]
Tests are added now

File: sqlglot/generators/bigquery.py [hunk 7]
```diff
@@ -688,3 +688,9 @@ def cast_sql(self, expression: exp.Cast, safe_prefix: str | None = None) -> str:
                 return f"{self.sql(expression, 'to')}{self.sql(this)}"
 
         return super().cast_sql(expression, safe_prefix=safe_prefix)
+
+    def clusterproperty_sql(self, expression: exp.ClusterProperty) -> str:
+        if expression.args.get("this"):
```
Comment (georgesittas) [inline sqlglot/generators/bigquery.py:693]:
```suggestion
        if expression.this:
```

File: sqlglot/generators/starrocks.py [hunk 8]
```diff
@@ -310,10 +310,13 @@ def partitionedbyproperty_sql(self, expression: exp.PartitionedByProperty) -> st
 
         return f"PARTITION BY {self.sql(this)}"
 
-    def cluster_sql(self, expression: exp.Cluster) -> str:
+    def clusterproperty_sql(self, expression: exp.ClusterProperty) -> str:
         """Generate StarRocks ORDER BY clause for clustering."""
+        if expression.args.get("this"):
```
Comment (georgesittas) [inline sqlglot/generators/starrocks.py:315]:
```suggestion
        if expression.this:
```

File: sqlglot/generator.py [hunk 9]
```diff
@@ -3042,6 +3043,13 @@ def withfill_sql(self, expression: exp.WithFill) -> str:
     def cluster_sql(self, expression: exp.Cluster) -> str:
         return self.op_expressions("CLUSTER BY", expression)
 
+    def clusterproperty_sql(self, expression: exp.ClusterProperty) -> str:
+        if expression.args.get("this"):
```
Comment (georgesittas) [inline sqlglot/generator.py:3047]:
```suggestion
        if expression.this:
```

**Review by geooo109 — APPROVED** [review]
@JoshuaM176 nice work! thanks
